### PR TITLE
ddl: Fix unit test `testPartitionCancelAddIndex`

### LINKED ISSUE #17145: db_partition_test.go:testIntegrationSuite4.TestPartitionCancelAddIndex failed
db_partition_test.go:testIntegrationSuite4.TestPartitionCancelAddIndex

```
[2020-05-12T06:51:00.930Z] ----------------------------------------------------------------------
[2020-05-12T06:51:00.930Z] FAIL: db_partition_test.go:1311: testIntegrationSuite4.TestPartitionCancelAddIndex
[2020-05-12T06:51:00.930Z]
[2020-05-12T06:51:00.930Z] db_partition_test.go:1314:
[2020-05-12T06:51:00.930Z]     testPartitionCancelAddIndex(c, s.store, s.dom.DDL(), s.lease, idxName, addIdxSQL)
[2020-05-12T06:51:00.930Z] /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/util/testkit/testkit.go:183:
[2020-05-12T06:51:00.930Z]     tk.c.Assert(err, check.IsNil, check.Commentf("sql:%s, %v, error stack %v", sql, args, errors.ErrorStack(err)))
[2020-05-12T06:51:00.930Z] ... value *errors.withStack = [domain:8028]Information schema is changed during the execution of the statement(for example, table definition may be updated by other DDL ran in parallel). If you see this error often, try increasing `tidb_max_delta_schema_count`. [try again later] ("[domain:8028]Information schema is changed during the execution of the statement(for example, table definition may be updated by other DDL ran in parallel). If you see this error often, try increasing `tidb_max_delta_schema_count`. [try again later]")
[2020-05-12T06:51:00.930Z] ... sql:delete from t1 where c1 = ?, [508], error stack [domain:8028]Information schema is changed during the execution of the statement(for example, table definition may be updated by other DDL ran in parallel). If you see this error often, try increasing `tidb_max_delta_schema_count`. [try again later]
```

 Latest failed builds: 
https://internal.pingcap.net/idc-jenkins/job/tidb_ghpr_unit_test/35955/display/redirect

### PR DESCRIPTION
**Pull request by djshow832**
### What problem does this PR solve?

Issue Number: close #17145

Problem Summary:
This test case is unstable because the DDL job and deletion are executing at the same time. Deletion may fail due to information schema changement.

### What is changed and how it works?

What's Changed:
Allow deletion and insertion fail. If they fail, the reason have to be information schema changement.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- N/A

### Release note 

- N/A

### CODE REVIEW

File: ddl/db_partition_test.go [hunk 1]
```diff
@@ -1891,11 +1891,13 @@ LOOP:
 			}
 			step := 10
 			rand.Seed(time.Now().Unix())
+			maxRetries := 3
```
Comment (AilinKid) [inline ddl/db_partition_test.go:1894]: does the following the SQL can always be successfully in 3 times? 
maybe the failpoint is better?

Comment (djshow832) [reply ddl/db_partition_test.go:1894]: It should succeed in the second time. DDL won't change so fast.
It's not good to let the deletion succeed when there's a DDL change. So failpoint is not good.

File: util/testkit/testkit.go [hunk 2]
```diff
@@ -326,3 +326,19 @@ func (tk *TestKit) GetTableID(tableName string) int64 {
...
 	return tbl.Meta().ID
 }
+
+// AssertErrorAndRetry asserts the error and retries the sql.
+func (tk *TestKit) AssertErrorAndRetry(errCode, maxRetries int, sql string, args ...interface{}) {
+	for i := 0; i <= maxRetries; i++ {
+		tk.c.Assert(i < maxRetries, check.IsTrue, check.Commentf("sql: %s, args: %v, retries: %d", sql, args, maxRetries))
+		res, err := tk.Exec(sql, args...)
+		if err != nil {
+			tk.MustGetErrCode(err.Error(), errCode)
```
Comment (tangenta) [inline util/testkit/testkit.go:336]: The first argument of `MustGetErrCode` is a SQL.

Comment (zimulala) [reply util/testkit/testkit.go:336]: @djshow832 Please address the comment.

**Pull request closed without merging**